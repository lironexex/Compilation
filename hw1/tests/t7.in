// Integers - DEC_REAL, HEX_FP:
var num1 = .2 + 8. - 0.2 * .8e-123 + 2.E+001
var num2 = 0xDeFp+2 -0xabcP+0 +0x123abcp-22
.2.234.232.256E+125
00xDefP-2
0Xabcp+123
