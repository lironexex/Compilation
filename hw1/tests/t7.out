1 COMMENT 1
2 VAR var
2 ID num1
2 ASSIGN =
2 DEC_REAL .2
2 BINOP +
2 DEC_REAL 8.
2 BINOP -
2 DEC_REAL 0.2
2 BINOP *
2 DEC_REAL .8e-123
2 BINOP +
2 DEC_REAL 2.E+001
3 VAR var
3 ID num2
3 ASSIGN =
3 HEX_FP 0xDeFp+2
3 BINOP -
3 HEX_FP 0xabcP+0
3 BINOP +
3 HEX_FP 0x123abcp-22
4 DEC_REAL .2
4 DEC_REAL .234
4 DEC_REAL .232
4 DEC_REAL .256E+125
5 DEC_INT 0
5 ID xDefP
5 BINOP -
5 DEC_INT 2
6 DEC_INT 0
6 ID Xabcp
6 BINOP +
6 DEC_INT 123
